wallets/core/src/wallet.ts

```diff
@@ -30,17 +35,19 @@ export interface Options {
 }
 
 class Wallet<InstanceType = any> {
+  public provider: InstanceType | null;
   private actions: WalletActions;
   private state: State;
   private options: Options;
-  private meta: BlockchainMeta[];
-  public provider: InstanceType | null;
+  private supportedBlockchains: BlockchainMeta[];
```

Comment:
I suggest make the contextual info (like `supportedBlockchains` and `isContractWallet`) into one object will let us to do the deprecations (if needed) or add/remove features more easily. In addition to that, the code will be easier to maintain I believe.

So what I think is something like this:
`private info: {
supportedBlockchains: BlockchainMeta[];
isContractWallet: boolean;
}`